File: app/src/main/java/com/rviannaoliveira/vchatfirebase/MessageAdapter.java
package com.rviannaoliveira.vchatfirebase;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.util.LinkedList;
import java.util.List;

/**
 * Criado por rodrigo on 28/05/17.
 */

public class MessageAdapter extends RecyclerView.Adapter<MessageAdapter.MessageViewHolder>{

    private List<ChatMessage> list = new LinkedList<>();
    private String user;

    MessageAdapter(String user){
        this.user = user;
    }

    void setMessageAdapter(List<ChatMessage> list){
        this.list.addAll(list);
        notifyDataSetChanged();
    }

    void setMessageAdapter(ChatMessage chatMessage){
        if(list.contains(chatMessage)){
            return;
        }
        this.list.add(chatMessage);
        notifyDataSetChanged();
    }

    @Override
    public MessageViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new MessageViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_message,parent,false));
    }

    @Override
    public void onBindViewHolder(MessageViewHolder holder, int position) {
        if(user.equals(list.get(position).getType())){
            holder.messageUser.setText(list.get(position).getMessage());
            holder.messageUser.setVisibility(View.VISIBLE);
        }else{
            holder.messageText.setText(list.get(position).getMessage());
            holder.messageText .setVisibility(View.VISIBLE);
        }
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class MessageViewHolder extends RecyclerView.ViewHolder{
        private TextView messageUser,messageText;

        MessageViewHolder(View itemView) {
            super(itemView);
            messageUser = (TextView) itemView.findViewById(R.id.message_user);
            messageText = (TextView) itemView.findViewById(R.id.message_text);
        }

    }
}

File: app/src/main/java/com/rviannaoliveira/vchatfirebase/MessageAdapter.kt
package com.rviannaoliveira.vchatfirebase

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import java.util.*

/**
 * Criado por rodrigo on 28/05/17.
 */

class MessageAdapter internal constructor(private val user: String) : RecyclerView.Adapter<MessageAdapter.MessageViewHolder>() {

    private val list = LinkedList<ChatMessage>()

    fun setMessageAdapter(chatMessage: ChatMessage) {
        if (list.contains(chatMessage)) {
            return
        }
        this.list.add(chatMessage)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MessageViewHolder {
        return MessageViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_message, parent, false))
    }

    override fun onBindViewHolder(holder: MessageViewHolder, position: Int) {
        if (user == list[position].type) {
            holder.messageUser.text = list[position].message
            holder.blockIn.visibility = View.VISIBLE
        } else {
            holder.messageText.text = list[position].message
            holder.blockOut.visibility = View.VISIBLE
        }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class MessageViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val messageUser: TextView = itemView.findViewById(R.id.message_user) as TextView
        val messageText: TextView = itemView.findViewById(R.id.message_text) as TextView
        val blockOut: LinearLayout = itemView.findViewById(R.id.block_out) as LinearLayout
        val blockIn: LinearLayout = itemView.findViewById(R.id.block_in) as LinearLayout

    }
}
